server/handlers/component_handler.go

```diff
@@ -1275,6 +1278,85 @@ func (h *Handler) RegisterMeshmodelComponents(rw http.ResponseWriter, r *http.Re
 	go h.config.MeshModelSummaryChannel.Publish()
 }
 
+// swagger:route POST /api/meshmodel/nonRegisterEntity
+// Handle POST request for registering meshmodel components.
+//
+// Validate the given value with the given schema
+// responses:
+// 	200:MeshModelHostsWithEntitySummary
+
+// request body should be json
+// request body should be of Host format
+
+func (handler *Handler) NonRegisterEntity(responseWriter http.ResponseWriter, request *http.Request, _ *models.Preference, user *models.User, provider models.Provider) {
```

Comment:
Still not according to API design.
@Jougan-0 The functionality that you are defining in the PR is crucial, while the output that we get from the current implementation is nice, the approach is not ideal.

How do you ensure that the registration might have been completed by the time UI makes a request to this endpoint?

My question still stands, if the server is doing the registration why you cannot consolidate all errors and send it all at once?
See the errorChan channel it listens to all the errors encountered, can't you send in more details on this channel `model, component and error` or the `event` object itself, store it in an array, and call the `cancel` function of the passed in context (when registration completes), when you will call the cancel func the ctx.Done case will be called, there you can iterate over that array and add in more details (i.e. summary, x model, y comps failed to register..... )

Reply:
[Screencast from 2024-01-09 06-22-53.webm](https://github.com/meshery/meshery/assets/99207534/5573a2e4-69de-411a-a91b-4442ee9069e2)
@MUzairS15 the reason for not using the errorChan is the same I am displaying in the video when the schema is nil the code works fine and these components are not registered but if we return an error such as schema empty the registration breaks I don't know the reason for it but this doesn't make sense i searched but i don't know why we exit quietly when the schema is empty and if we return an error the code breaks so i opt for using the meshkit to store the error and store their in meshkit itself when the entites try to register .

Reply:
API design is still not accurate, invoking the event from UI is also inaccurate, find a way to send events from the server.